bnd/derby.bnd

```diff
@@ -0,0 +1,106 @@
+-classpath:       ../jars/sane/derby.jar
+-include:         ../classes/bnd/version.properties, locales.properties
+-includeresource: @derby.jar!/META-INF/**, @derby.jar!/module-info.class
+
+-output:       ../classes/bnd/derby.jar
+-savemanifest: ../classes/bnd/derby.mf
+
+Class-Path: derbyshared.jar ${localeList}
+
+#
+# Specify that the minimum Java requirement is Java 8. We specify "-noee"
+# so that bnd doesn't try to infer the requirement from the bytecode (because
+# module descriptors will cause it to believe that Java 9 is a hard
+# requirement).
+#
+-noee: true
+Require-Capability: osgi.ee;filter:="(&(osgi.ee=JavaSE)(version=1.8))"
+
+#
+# Remove headers that reduce reproducibility.
+#
+-noextraheaders: true
+
+#
+# Specify basic OSGi metadata.
+#
+Bundle-DocURL:       https://db.apache.org/derby/manuals/index.html
+Bundle-Name:         Apache Derby Embedded Engine ${major}.${minor}
+Bundle-SymbolicName: org.apache.derby.engine
+Bundle-Vendor:       Apache Software Foundation
+Bundle-Version:      ${major}.${minor}.${maint}
+
+Bundle-Activator:    org.apache.derby.osgi.EmbeddedActivator
+
+#
+# Specify that all packages are private by default, and sealed. We will
+# override the privacy of an explicit list of packages by exporting them
+# with Export-Package.
+#
+Private-Package: *
+Sealed:          true
+
+Import-Package:\
+javax.naming;resolution:="optional", \
+javax.naming.directory;resolution:="optional", \
+javax.management;resolution:="optional", \
+javax.transaction.xa;resolution:="optional", \
+*
+
+Export-Package: \
+org.apache.derby.agg, \
```

Comment:
I wonder if an `Export-Package: org.apache.derby.*` would be sufficient here.

Reply:
As for checking bnd into version control: I would normally never do this, but it seems that the project is already doing this for pretty much all of the build tools.

For `Export-Package`: If I remember correctly, there are packages that are not exported by the JPMS module definition. Exporting a package with OSGi that's not exported by the VM's module definition will result in trouble, won't it?

Reply:
Okay, maybe there are reasons why these packages are omitted, it just feels a lot of maintenance need to keep them in sync, but maybe this is for historical reasons.
I think you are right about the checked in tools didn't noticed that, maybe it would be good to use https://ant.apache.org/ivy/ instead, but that's a little bit out of scope here.
Lets see what the maintainers think about this.

Reply:
> Okay, maybe there are reasons why these packages are omitted, it just feels a lot of maintenance need to keep them in sync

Yep. What I'm hoping is that the OSGi/Bnd people will come up with a way to calculate OSGi package exports from existing JPMS descriptors (if descriptors are present). This would eliminate the duplication.